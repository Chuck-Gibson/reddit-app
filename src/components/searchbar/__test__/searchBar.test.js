import { create } from "react-test-renderer";
import { screen, cleanup, fireEvent, render } from "@testing-library/react";
import SearchBar from "../searchBar";

afterEach(() => cleanup);
beforeEach(() => render(<SearchBar />));

describe("Search bar integration tests", () => {
  it("should render input and submit", () => {
    const buttonElement = screen.getByRole("button");
    const inputElement = screen.getByRole("search");

    expect(buttonElement).toBeInTheDocument();
    expect(inputElement).toBeInTheDocument();
  });

  it("should clear the input field when submit button is clicked", () => {
    const buttonElement = screen.getByRole("button");
    const placeholderText = screen.getByPlaceholderText("Search something...");
    const userInput = screen.getByRole("search");

    // Type search queries
    fireEvent.change(userInput, {
      target: {
        value: "hello",
      },
    });
    // Set proper value
    expect(userInput.value).toBe("hello");
    // Click search button
    fireEvent.click(buttonElement);
    // Clear input
    expect(userInput.value).toBe("");
    // Place holder text visible
    expect(placeholderText).toBeVisible();
  });
});

describe("Search bar snapshot testing", () => {
  it("should render with no value", () => {
    const treeNodes = create(<SearchBar />);

    expect(treeNodes.toJSON()).toMatchSnapshot();
  });
});
