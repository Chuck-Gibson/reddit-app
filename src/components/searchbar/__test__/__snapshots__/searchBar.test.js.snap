// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar snapshot testing should render with no value 1`] = `
<div
  className="SearchBar"
>
  <input
    onChange={[Function]}
    placeholder="Search something..."
    role="search"
    type="text"
    value=""
  />
  <button
    className={null}
    onClick={[Function]}
  >
    <svg>
      search.svg
    </svg>
  </button>
</div>
`;
